examples/vulnerabilities-connector/app/VulnerabilitiesConnector.go

```diff
@@ -0,0 +1,320 @@
+}
+
+func (v *VulnerabilitiesConnector) Initialize() {
+	// Tenants Config Folder
+	_, err := os.Stat("./tenant_config_files")
+	if err != nil && errors.Is(err, fs.ErrNotExist) {
+		os.Mkdir("./tenant_config_files", 0700)
+	}
+
+	// Init
+	v.Config = InitConfig()
+	v.Providers = []ProviderConfiguration{}
+
+	log.SetFlags(log.Llongfile | log.Ldate | log.Ltime)
+	v.Logger = log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime)
+
+	// Providers
+	v.Providers = v.buildVulnerabilityProvidersConfig()
+
+	v.Logger.Println("Example initialized...")
+}
+
+func (v *VulnerabilitiesConnector) QueryAssets(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	filter := []*string{}
+
+	assets, err := client.Vulnerabilities.QueryAssets(ctx, &engine.QueryAssetsRequest{
+		Filter: filter,
+	})
+	if err != nil {
+		v.Logger.Printf("Error executing QueryAssets for %s provider: %s\n", provider, err.Error())
+		return
+	}
+	v.Logger.Printf("Found %d security assets from %s\n", len(assets.Result), provider.ProviderConfig.Type)
+}
+
+func (v *VulnerabilitiesConnector) QueryFindings(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	filter := []*string{
+		engine.String("severity[in]Critical"),
+	}
+
+	findings, err := client.Vulnerabilities.QueryFindings(ctx, &engine.QueryFindingsRequest{
+		Filter: filter,
+	})
+	if err != nil {
+		v.Logger.Printf("Error executing QueryFindings for %s provider: %s\n", provider, err.Error())
+		return
+	}
+	v.Logger.Printf("Found %d security findings from %s\n", len(findings.Result), provider.ProviderConfig.Type)
+}
+
+func (v *VulnerabilitiesConnector) QueryScans(ctx context.Context, client *engineClient.Client, provider *mgmt.CreateIntegrationRequest) {
+	var e *engine.NotImplementedError
+
+	assets, err := client.Vulnerabilities.QueryScans(ctx, &engine.QueryScansRequest{})
+	if err != nil && !errors.As(err, &e) {
+		v.Logger.Printf("Error executing QueryScans for %s provider: %s\n", provider, err.Error())
+		return
+	} else if errors.As(err, &e) {
+		// not all providers support the QueryScans operation. If an operation is not supported, the API will return
+		// a NotImplementedError so we can check for that error and handle it specifically
+
+		v.Logger.Printf("Query scans not implemented for %s\n", provider.ProviderConfig.Type)
+		return
+	}
+
+	v.Logger.Printf("Found %d scans from %s\n", len(assets.Result), provider.ProviderConfig.Type)
+}
+
+func (v *VulnerabilitiesConnector) Start(cleanUp bool) {
+	v.Logger.Printf("Using %d vulnerability providers\n", len(v.Providers))
+
+	/*ctx := context.Background()
+
+	for _, provider := range v.Providers {
+		client := provider.Tenant.Synqly.EngineClients["vulnerabilities"]
+
+		v.QueryAssets(ctx, client, provider.IntegrationRequest)
+		v.QueryFindings(ctx, client, provider.IntegrationRequest)
+		v.QueryScans(ctx, client, provider.IntegrationRequest)
+	}*/
+
+	if cleanUp {
+		defer v.CleanUp()
+	}
```

Comment:
This performs the cleanup and the end of this method. Instead, can you have this return a `func()`? 

then here you could do 

```
if cleanUp {
  return v.CleanUp()
}

return func(){}
```

then you will need to call the returned method in main.go